pkgs/development/compilers/factor-lang/factor98.nix

```diff
@@ -0,0 +1,178 @@
+{ stdenv, lib, git, curl, makeWrapper, fetchurl, unzip, runCommand, writeScriptBin
+, interpreter
+, glib
+, pango
+, cairo
+, gtk2-x11
+, gdk-pixbuf
+, gnome2
+, pcre
+, libGL
+, libGLU
+, freealut
+, openssl
+, udis86
+, openal
+, libogg
+, libvorbis
+, graphviz
+, librsvg
+, zlib
+, tzdata
+, ncurses
+}:
+let
+  runtimeLibs = [ glib pango cairo gtk2-x11 gdk-pixbuf gnome2.gtkglext pcre
+    libGL libGLU freealut openssl udis86 openal libogg libvorbis graphviz zlib ];
+
+  wrapFactorSh = { from, to ? false, runtimeLibs }: ''
+    # Set Gdk pixbuf loaders file to the one from the build dependencies here
+    unset GDK_PIXBUF_MODULE_FILE
+    # Defined in gdk-pixbuf setup hook
+    findGdkPixbufLoaders "${librsvg}"
+
+    ${if to then "makeWrapper ${from} ${to}" else "wrapProgram ${from}"} \
+      --set GDK_PIXBUF_MODULE_FILE "$GDK_PIXBUF_MODULE_FILE" \
+      --argv0 factor \
+      --prefix LD_LIBRARY_PATH : /run/opengl-driver/lib:${lib.makeLibraryPath runtimeLibs} \
+      --prefix PATH : ${lib.getBin graphviz}/bin
+  '';
+
+  wrapFactor = runtimeLibs:
+    runCommand (lib.appendToName "with-libs" interpreter).name {
+      nativeBuildInputs = [ makeWrapper gdk-pixbuf ];
```

Comment:
You are sure gdk-pixbuf does not go to buildInputs?

Reply:
I am using its setup-hook function and was not sure, whether I have to use the compile-time or run-time package for that.

Reply:
I am using its setup-hook function and wasn't sure, whether to use the compile-time or run-time package.